metabase/metabase
# Remove some console error

## Changes
1. Remove prop type error in `<CopyWidget />`
1. Remove deprecated lifecycle name warning by upgrading `react-transition-group`
1. Fix modal transition not working in some cases

## Benefits
This will help troubleshooting unit tests easier since we're removing some console errors and warnings which eat up the output screen real estate when running tests.

## How to verify
1. Open question, or dashboard sharing modal, there should not see the prop type console error about not specifying `readOnly`
1. When opening any modal, there should be no console warning about the deprecated React component lifecycle.
1. When saving a new question, the fade-in transition should work.
1. When editing a question, click `Save as new question` and the transition should work
    <img width="668" alt="image" src="https://user-images.githubusercontent.com/1937582/216694262-d4853b2a-cb72-466d-8b38-6a2ecad22370.png">
1. Go to http://localhost:3000/reference/segments, or http://localhost:3000/reference/metrics and click on any existing item there, you should create one if there is none. Click formula section and the transition should work.
    <img width="691" alt="image" src="https://user-images.githubusercontent.com/1937582/216694731-90e81601-9a33-48b2-90a0-cc4c8f7c637b.png">


## Demo
### Before
- <img width="1450" alt="Screenshot 2023-02-03 at 7 31 16 PM" src="https://user-images.githubusercontent.com/1937582/216692467-c73368e6-13fb-49e7-81a7-94333674bdd0.png">
- <img width="1445" alt="Screenshot 2023-02-03 at 7 28 57 PM" src="https://user-images.githubusercontent.com/1937582/216692409-ff586a02-e53b-4e86-b741-e9134c61c69a.png">
- ![before modal fix](https://user-images.githubusercontent.com/1937582/216693260-9e2d5d96-ccf6-45b4-be7e-7d19f7465f73.gif)

### After
- <img width="1445" alt="Screenshot 2023-02-03 at 7 32 56 PM" src="https://user-images.githubusercontent.com/1937582/216692552-2683fa5a-d885-4bfd-bb20-234f5c609c0d.png">
- <img width="1444" alt="Screenshot 2023-02-03 at 7 33 22 PM" src="https://user-images.githubusercontent.com/1937582/216692568-eb2f541a-b85f-4a45-b3c0-35a31ee8eee6.png">
- ![after modal fix](https://user-images.githubusercontent.com/1937582/216693377-15f5f125-7ace-4727-8e28-f9092a65d73a.gif)
